File: tests/swing_high_test.py
import numpy as np

from taflow.swing_high import SwingHigh


def test_swing_high_lifecycle_and_reset():
    state = SwingHigh(np.array([], dtype=float), np.array([], dtype=float), swing_length=2)
    state.extend([10, 11, 12, 11, 10], [8, 7, 6, 7, 8])
    assert len(state) == 5
    state.reset()
    assert state.value is None

File: crates/taflow-core/src/stream/swing_high_test.rs
use super::swing_high::SwingHigh;

#[test]
fn lifecycle_and_reset_are_consistent() {
    let mut state = SwingHigh::new(2).unwrap();
    state.append(10.0, 8.0);
    state.reset();
    assert!(state.value().is_none());
}
